pkg/owner/manager.go

```diff
@@ -364,13 +364,13 @@ func (m *ownerManager) campaignLoop(campaignContext context.Context) {
 		select {
 		case <-m.etcdSes.Done():
 			m.logger.Info("etcd session done, refresh it")
-			if err2 := m.refreshSession(util2.NewSessionRetryUnlimited, ManagerSessionTTL); err2 != nil {
+			if err2 := m.refreshSession(campaignContext, util2.NewSessionRetryUnlimited, ManagerSessionTTL); err2 != nil {
```

Comment:
This is the core of the fix.
We should pass the `campaignContext`.

The code is:

```
campaignContext, cancel := context.WithCancel(m.ctx)
go campaignLoop(campaignContext)
cancel()
```

If we use `m.ctx` in the campaign loop, `cancel()` cannot cancel  the ctx!